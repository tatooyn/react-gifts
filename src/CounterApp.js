import React, { useState } from 'react';
import PropTypes from 'prop-types';
const CounterApp= ({valueProp=10})=>{
    const [value, setValue] = useState(valueProp)
    const masUno=()=>{setValue(value+1)};
    const menosUno=()=>{setValue(value-1)};
    const reset=()=>{setValue(valueProp)};
    return (
        <>
            <h1>CounterApp</h1>
            <h1>{value}</h1>
            <button onClick={menosUno}>-1</button>
            <button onClick={reset}>Reset</button>
            <button onClick={masUno}>+1</button>
        </>
    );
};
CounterApp.propTypes={
    value: PropTypes.number
}

export default CounterApp;