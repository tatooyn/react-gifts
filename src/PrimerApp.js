import React from 'react';
import PropTypes from 'prop-types';
const PrimerApp= ({saludo})=>{
    return (
        <>
            <h1>{saludo}!</h1>
            <h2>Â¡Como estas!</h2>
        </>
    );
};
PrimerApp.propTypes={
    saludo: PropTypes.string.isRequired
}

export default PrimerApp;