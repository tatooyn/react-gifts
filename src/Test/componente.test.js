import React from 'react';
import { shallow } from "enzyme";
import PrimerApp from "../PrimerApp";
import CounterApp from '../CounterApp';

describe('==Componentes==',()=>{
    beforeEach(()=>{
        const saludo='Hola soy Andres!';
        const component=shallow(<PrimerApp saludo={saludo}/>);
    })
    test('componente',()=>{
        expect(component).toMatchSnapshot();
    })
    test('html',()=>{
        const saludo='Hola soy Andres!';
        // const component=shallow(<PrimerApp saludo={saludo}/>);
        const tituloTest=component.find('h1').text();
        expect(tituloTest).toBe(`${saludo}!`);
    })
    test('click',()=>{
        const valueTest=10;
        const component=shallow(<CounterApp/>);
        component.find('button').at(0).simulate('click');
        const valor=component.find('h1').at(1).text();
        expect(valor).toBe(`${valueTest-1}`);
    })
})
