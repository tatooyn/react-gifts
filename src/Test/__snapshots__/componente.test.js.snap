// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`==Componentes== componente 1`] = `
<Fragment>
  <h1>
    Hola soy Andres!
    !
  </h1>
  <h2>
    Â¡Como estas!
  </h2>
</Fragment>
`;
